src/protocol.cpp

```diff
@@ -263,45 +226,59 @@ CInv::CInv(int typeIn, const uint256& hashIn)
     hash = hashIn;
 }
 
-CInv::CInv(const std::string& strType, const uint256& hashIn)
-{
-    unsigned int i;
-    for (i = 1; i < ARRAYLEN(ppszTypeName); i++)
-    {
-        if (strType == ppszTypeName[i])
-        {
-            type = i;
-            break;
-        }
-    }
-    if (i == ARRAYLEN(ppszTypeName))
-        throw std::out_of_range(strprintf("CInv::CInv(string, uint256): unknown type '%s'", strType));
-    hash = hashIn;
-}
-
 bool operator<(const CInv& a, const CInv& b)
 {
     return (a.type < b.type || (a.type == b.type && a.hash < b.hash));
 }
 
 bool CInv::IsKnownType() const
 {
-    return (type >= 1 && type < (int)ARRAYLEN(ppszTypeName));
+    return GetCommandInternal() != nullptr;
+}
+
+const char* CInv::GetCommandInternal() const
+{
+    switch (type)
+    {
+        case MSG_TX:                        return NetMsgType::TX;
+        case MSG_BLOCK:                     return NetMsgType::BLOCK;
+        case MSG_FILTERED_BLOCK:            return NetMsgType::MERKLEBLOCK;
+        case MSG_TXLOCK_REQUEST:            return NetMsgType::TXLOCKREQUEST;
+        case MSG_TXLOCK_VOTE:               return NetMsgType::TXLOCKVOTE;
+        case MSG_CMPCT_BLOCK:               return NetMsgType::CMPCTBLOCK;
+        case MSG_SPORK:                     return NetMsgType::SPORK;
+        case MSG_DSTX:                      return NetMsgType::DSTX;
+        case MSG_GOVERNANCE_OBJECT:         return NetMsgType::MNGOVERNANCEOBJECT;
+        case MSG_GOVERNANCE_OBJECT_VOTE:    return NetMsgType::MNGOVERNANCEOBJECTVOTE;
+        case MSG_QUORUM_FINAL_COMMITMENT:   return NetMsgType::QFCOMMITMENT;
+        case MSG_QUORUM_CONTRIB:            return NetMsgType::QCONTRIB;
+        case MSG_QUORUM_COMPLAINT:          return NetMsgType::QCOMPLAINT;
+        case MSG_QUORUM_JUSTIFICATION:      return NetMsgType::QJUSTIFICATION;
+        case MSG_QUORUM_PREMATURE_COMMITMENT: return NetMsgType::QPCOMMITMENT;
+        case MSG_QUORUM_RECOVERED_SIG:      return NetMsgType::QSIGREC;
+        case MSG_CLSIG:                     return NetMsgType::CLSIG;
+        case MSG_ISLOCK:                    return NetMsgType::ISLOCK;
+        default:
+            return nullptr;
+    }
 }
 
-const char* CInv::GetCommand() const
+std::string CInv::GetCommand() const
 {
-    if (!IsKnownType())
+    auto cmd = GetCommandInternal();
+    if (!cmd) {
```

Comment:
Hmm...what part do you find hard to read and what would you prefer?

Reply:
Likely change to `

Reply:
But `cmd` is not a `std::string` but a `const char*`

Reply:
At least `pointer==nullptr` instead of `!pointer`

Reply:
I'll change it in this instance, but generally this is something that is sprinkled all over the source already, simply because it's quite typical for C++ code, independent from it being bad style or not.